junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,447 @@
+          hasAnnotation[org.junit.Before](ddef) ||
+          hasAnnotation[org.junit.After](ddef) ||
+          hasAnnotation[org.junit.BeforeClass](ddef) ||
+          hasAnnotation[org.junit.AfterClass](ddef) ||
+          hasAnnotation[org.junit.Ignore](ddef)
+      }
+
+      def hasAnnotation[T: TypeTag](ddef: DefDef): Boolean =
+        ddef.symbol.annotations.exists(_.atp == typeOf[T])
+
+      // Copy of gen.mkClassDef for 2.10 compat
+      private def gen_mkClassDef_compat(mods: Modifiers, name: TypeName,
+          tparams: List[TypeDef], templ: Template): ClassDef = {
+        val isInterface = mods.isTrait &&
+          (templ.body forall treeInfo.isInterfaceMember)
+        val mods1 = if (isInterface) mods | Flags.INTERFACE else mods
+        ClassDef(mods1, name, tparams, templ)
+      }
+
+      // Copy of newValDef for 2.10 compat
+      private def newValDef_compat(sym: Symbol, rhs: Tree)(
+        mods: Modifiers = Modifiers(sym.flags),
+        name: TermName = sym.name.toTermName,
+        tpt: Tree = TypeTreeMemberType(sym)): ValDef = {
+        atPos(sym.pos)(ValDef(mods, name, tpt, rhs)) setSymbol sym
+      }
+
+      // Copy of newDefDef for 2.10 compat
+      private def newDefDef_compat(sym: Symbol, rhs: Tree)(
+          mods: Modifiers = Modifiers(sym.flags),
+          name: TermName = sym.name.toTermName,
+          tparams: List[TypeDef] = sym.typeParams.map(sym =>
+            newTypeDef_compat(sym, typeBoundsTree(sym))()),
+          vparamss: List[List[ValDef]] = mapParamss(sym)(sym =>
+            newValDef_compat(sym, EmptyTree)()),
+          tpt: Tree = TypeTreeMemberType(sym)): DefDef = {
+        atPos(sym.pos)(DefDef(mods, name, tparams, vparamss, tpt, rhs)).setSymbol(sym)
+      }
+
+      // Copy of newTypeDef for 2.10 compat
+      def newTypeDef_compat(sym: Symbol, rhs: Tree)(
+          mods: Modifiers = Modifiers(sym.flags),
+          name: TypeName = sym.name.toTypeName,
+          tparams: List[TypeDef] = sym.typeParams.map(sym =>
+            newTypeDef_compat(sym, typeBoundsTree(sym))())): TypeDef = {
+        atPos(sym.pos)(TypeDef(mods, name, tparams, rhs)) setSymbol sym
+      }
+
+      // Copy of TypeTreeMemberType for 2.10 compat
+      private def TypeTreeMemberType(sym: Symbol): TypeTree = {
+        val resType = {
+          if (sym.owner.isTerm) sym.tpe
+          else sym.owner.thisType.memberType(sym)
+        }.finalResultType
+        atPos(sym.pos.focus)(TypeTree(resType))
+      }
+
+      // Copy of TypeBoundsTree for 2.10 compat
+      private def typeBoundsTree(bounds: TypeBounds): TypeBoundsTree =
+        TypeBoundsTree(TypeTree(bounds.lo), TypeTree(bounds.hi))
+
+      // Copy of TypeBoundsTree for 2.10 compat
+      private def typeBoundsTree(sym: Symbol): TypeBoundsTree =
+        atPos(sym.pos)(typeBoundsTree(sym.info.bounds))
+
+      private def termNames_CONSTRUCTOR: TermName = {
+        // termNames.CONSTRUCTOR
+        newTermName("<init>")
```

Comment:
There should be `nme.CONSTRUCTOR` or something similar where this is precomputed.